gluoncv/model_zoo/rcnn/rcnn.py

```diff
@@ -151,6 +151,19 @@ def set_nms(self, nms_thresh=0.3, nms_topk=400, post_nms=100):
         self.nms_topk = nms_topk
         self.post_nms = post_nms
 
+    def reset_class(self, classes):
+        """Reset class categories and class predictors.
+
+        Parameters
+        ----------
+        classes : iterable of str
+            The new categories. ['apple', 'orange'] for example.
+
+        """
+        self.classes = classes
+        self.num_class = len(classes)
+        self.class_predictor = nn.Dense(self.num_class + 1, weight_initializer=mx.init.Normal(0.01))
```

Comment:
Also need to reset box_predictor, cls_decoder, target_generator

Reply:
And call _clear_cached_op as is in set_nms.

Reply:
The predictor was created with name_scope so it might be best reset with name_scope as well.